### PR TITLE
use system resources more intelligently

### PR DESCRIPTION
**Pull request by tkellen**

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -1,14 +1,34 @@
 # -*- mode: ruby -*-
 # vi: set ft=ruby :
 
-VAGRANTFILE_API_VERSION = '2'
+VAGRANTFILE_API_VERSION = "2"
```
Comment (cowboy) [inline Vagrantfile]: This line didn't need to change!

File: Vagrantfile [hunk 2]
```diff
@@ -1,14 +1,34 @@
...
+
+# look up system cpu and ram so we can use more intelligent defaults
+LINUX = RUBY_PLATFORM =~ /linux/
+OSX = RUBY_PLATFORM =~ /darwin/
+WINDOWS = (/cygwin|mswin|mingw|bccwin|wince|emx/ =~ RUBY_PLATFORM) != nil
+if OSX
+  CPUS = `sysctl -n hw.ncpu`.to_i
+  MEM = `sysctl -n hw.memsize`.to_i / 1024 / 1024 / 4
+end
+if LINUX
```
Comment (cowboy) [inline Vagrantfile]: `elsif`

File: Vagrantfile [hunk 3]
```diff
@@ -1,14 +1,34 @@
...
+  CPUS = `nproc`.to_i
+  MEM = `sed -n -e '/^MemTotal/s/^[^0-9]*//p' /proc/meminfo`.to_i / 1024 / 4
+end
+if WINDOWS
```
Comment (cowboy) [inline Vagrantfile]: `elsif`

Comment (cowboy) [inline Vagrantfile]: I don't think we're supporting Windows. Is this case necessary?

**Comment by tkellen** [comment]
Updated!

File: Vagrantfile [hunk 4]
```diff
@@ -3,12 +3,27 @@
 
 VAGRANTFILE_API_VERSION = '2'
 
+# look up system cpu and ram so we can use more intelligent defaults
+LINUX = RUBY_PLATFORM =~ /linux/
+OSX = RUBY_PLATFORM =~ /darwin/
+WINDOWS = (/cygwin|mswin|mingw|bccwin|wince|emx/ =~ RUBY_PLATFORM) != nil
```
Comment (cowboy) [inline Vagrantfile]: Remove this line?

File: Vagrantfile [hunk 5]
```diff
@@ -18,6 +33,12 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
   # sudo). This should match the Ansible host_vars/vagrant site_fqdn value.
   config.hostsupdater.aliases = ['deployment-workflow.loc']
 
+  # give vm access to 1/4 total system memory and all cpu
+  config.vm.provider "virtualbox" do |v|
```
Comment (cowboy) [inline Vagrantfile]: Single quotes in this block?

**Comment by tkellen** [comment]
Updated again.

**Comment by tkellen** [comment]
We good now @cowboy?

**Comment by cowboy** [comment]
Seems good to me, I haven't been able to test it, but I trust that you have!

**Comment by tkellen** [comment]
Yup, we've been using this pattern on the API vagrantfile for some time now.

before:
screen shot 2015-11-23 at 10 51 54 am

after:
screen shot 2015-11-23 at 10 51 07 am

**Pull request merged**